packages/in_app_purchase/lib/src/in_app_purchase_connection/google_play_connection.dart

```diff
@@ -59,4 +60,33 @@ class GooglePlayConnection
       _billingClient.startConnection(onBillingServiceDisconnected: () {});
 
   Future<void> _disconnect() => _billingClient.endConnection();
+
+  /// Query the product detail list.
+  ///
+  /// This method only returns [QueryProductDetailsResponse].
+  /// To get detailed Google Play sku list, use [BillingClient.querySkuDetails]
+  /// to get the [SkuDetailsResponseWrapper].
+  Future<QueryProductDetailsResponse> queryProductDetails(
+      Set<String> identifiers) async {
+    SkuDetailsResponseWrapper inappResponse =
+        await _billingClient.querySkuDetails(
+            skuType: SkuType.inapp, skusList: identifiers.toList());
+    SkuDetailsResponseWrapper subResponse = await _billingClient
+        .querySkuDetails(skuType: SkuType.subs, skusList: identifiers.toList());
+    List<ProductDetails> inappProducts = inappResponse.skuDetailsList
```

Comment:
nit, subjective: instead of writing this conversion twice what about combining these lists first?

Something like:

```
List<ProductDetails> products = (inappResponse.skuDetailsList + subResponse.skuDetailsList)
    .map((SkuDetailsWrapper wrapper) => //etc
```